### PR TITLE
Deprecate the use of ParamConverter in favor of argument resolver

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Instead of relying on `SensioFrameworkExtraBundle` `ParamConverter` feature, we can do the same thanks to value resolvers: https://symfony.com/doc/4.4/controller/argument_value_resolver.html and remove the dependency of that bundle.

This would only affect to people using `@ParamConverter` annotation, for those type-hinting with the admin classes should work as it was.

I am targeting this branch, because these changes are BC.

## Changelog

```markdown
### Deprecated
- Deprecated using `@ParamConverter` annotation for injecting admin services
```

### CODE REVIEW

File: src/ArgumentResolver/AdminValueResolver.php
Comment (VincentLanglet) [inline src/ArgumentResolver/AdminValueResolver.php:52]: `$this->adminFetcher->get($request)` is also done in the `supports` method. No perf impact ?

File: src/Request/ParamConverter/AdminParamConverter.php
Comment (VincentLanglet) [inline src/Request/ParamConverter/AdminParamConverter.php:25]: Should we trigger deprecation somewhere ?

**Comment by VincentLanglet** [comment]
@sonata-project/contributors I wanted to release a Release candidate of Sonata 4 soon.
If we still want to introduce deprecation (like this one or maybe #7309), either we should merge it quick, either I should release an alpha-3 instead of a RC. WDYT ?

**Comment by jordisala1991** [comment]
Feature freeze or we always have something little thing to add

File: src/Request/ParamConverter/AdminParamConverter.php
Comment (franmomu) [reply src/Request/ParamConverter/AdminParamConverter.php:25]: To be honest, I don't know if we can check if the controller is type-hinted or not given the `ParamConverter`.

**Comment by franmomu** [comment]
https://github.com/sonata-project/SonataAdminBundle/pull/7364 is also adding a feature AFAIK

File: src/ArgumentResolver/AdminValueResolver.php
Comment (franmomu) [reply src/ArgumentResolver/AdminValueResolver.php:52]: I wanted to avoid failing when resolving, that's why I moved all the checks to the `supports` method.

About performance, this is supposed to be executed only once per request and the `Pool` container should already have the admin instantiated, I would say it is not a big deal.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
> #7364 is also adding a feature AFAIK

I kinda agree. And that's why I was waiting for the RC. I opened this issue #7349 for 2 weeks and was just waiting for approval. But we need to stop at one moment and make the release of it will never be done. I'm not against merging this at all since it seems ready.
It was just a more general question "Can we feature-freeze now or should we wait for something else ? If we wait for something else, can it be done quickly ?". 

If it's ok for everybody, let's just say we
- merge https://github.com/sonata-project/SonataAdminBundle/pull/7364 and this.
- make a last 3.x release
- merge 3.x into master
- make the 4-RC release

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**
